Golem/Yagna/YagnaApi.cs

```diff
@@ -0,0 +1,276 @@
+                ?? throw new Exception($"Failed to deserialize REST call reponse to type: {typeof(T).Name}");
+        }
+
+        public async Task<MeInfo> Me(CancellationToken token)
+        {
+            return await RestCall<MeInfo>("/me", token);
+        }
+
+        public async Task<YagnaAgreement> GetAgreement(string agreementId, CancellationToken token = default)
+        {
+            return await RestCall<YagnaAgreement>($"/market-api/v1/agreements/{agreementId}", token);
+        }
+
+        public async Task<List<YagnaAgreement>> GetAgreements(CancellationToken token = default)
+        {
+            return await RestCall<List<YagnaAgreement>>($"/market-api/v1/agreements", token);
+        }
+
+        public async Task<ActivityStatePair> GetState(string activityId, CancellationToken token = default)
+        {
+            return await RestCall<ActivityStatePair>($"/activity-api/v1/activity/{activityId}/state", token);
+        }
+
+        public async Task<string> GetActivityAgreement(string activityId, CancellationToken token = default)
+        {
+            return await RestCall<string>($"/activity-api/v1/activity/{activityId}/agreement", token);
+        }
+
+        public async IAsyncEnumerable<TrackingEvent> ActivityMonitorStream([EnumeratorCancellation] CancellationToken token = default)
+        {
+            await foreach (var item in RestStream<TrackingEvent>($"/activity-api/v1/_monitor", token))
+            {
+                yield return item;
+            }
+        }
+
+        public async Task<List<string>> GetActivities(CancellationToken token = default)
+        {
+            var list =  await RestCall<List<string>>($"/activity-api/v1/activity", token);
+            return list;
+        }
+
+        public async Task<List<Invoice>> GetInvoices(DateTime? afterTimestamp = null, CancellationToken token = default)
+        {
+            var path = "/payment-api/v1/invoices";
+            if(afterTimestamp != null)
+                path += new QueryBuilder(new Dictionary<string, string>{
+                    {"after_timestamp", afterTimestamp.Value.ToUniversalTime().ToString()}
+                });
+            var list =  await RestCall<List<Invoice>>(path, token);
+            return list;
+        }
+
+        public async Task<List<Payment>> GetPayments(DateTime? afterTimestamp = null, CancellationToken token = default)
+        {
+            var path = "/payment-api/v1/payments";
+            if(afterTimestamp != null)
+                path += new QueryBuilder(new Dictionary<string, string>{
+                    {"after_timestamp", afterTimestamp.Value.ToUniversalTime().ToString()}
+                });
+            var list =  await RestCall<List<Payment>>(path, token);
+            return list;
+        }
+
+        public async Task<Invoice?> GetInvoice(string id, CancellationToken token)
+        {
+            Invoice? invoice = null;
```

Comment:
I would prefer function throwing exception. If you need to get null you can create another function that catches exception and returns null